File: app/src/main/java/com/sygic/travel/sdkdemo/filters/CategoriesAdapter.java
package com.sygic.travel.sdkdemo.filters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.sygic.travel.sdkdemo.R;

public class CategoriesAdapter extends RecyclerView.Adapter<CategoriesAdapter.ViewHolder> {
	private static String[] categoriesKeys = {"all", "sightseeing", "shopping", "eating", "discovering", "playing",
		"traveling", "going_out", "hiking", "sports", "relaxing"};

	private static String[] categoriesNames = {"All categories", "Sightseeing", "Shopping", "Restaurants", "Museums", "Family",
		"Transport", "Nightlife", "Outdoors", "Sports", "Relaxation"};

	private ViewHolder.CategoryClick categoryClick;

	public CategoriesAdapter(ViewHolder.CategoryClick categoryClick) {
		this.categoryClick = categoryClick;
	}

	@Override
	public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		View vItem = LayoutInflater
			.from(parent.getContext())
			.inflate(R.layout.category_item, parent, false);
		return new ViewHolder(vItem, categoryClick);
	}

	@Override
	public void onBindViewHolder(ViewHolder holder, int position) {
		holder.tvCategory.setText(categoriesNames[position]);
	}

	@Override
	public int getItemCount() {
		if(categoriesNames == null) {
			return 0;
		}
		return categoriesNames.length;
	}

	public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
		private TextView tvCategory;
		private CategoryClick categoryClick;

		public ViewHolder(View vItem, CategoryClick categoryClick) {
			super(vItem);
			this.categoryClick = categoryClick;
			tvCategory = (TextView) vItem;
			tvCategory.setOnClickListener(this);
		}

		@Override
		public void onClick(View tvCategory) {
			categoryClick.onCategoryClick(
				categoriesKeys[getAdapterPosition()],
				categoriesNames[getAdapterPosition()]
			);
		}

		public interface CategoryClick{
			void onCategoryClick(String categoryKey, String categoryName);
		}
	}
}

File: app/src/main/java/com/sygic/travel/sdkdemo/filters/CategoriesAdapter.kt
package com.sygic.travel.sdkdemo.filters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.sygic.travel.sdkdemo.R

class CategoriesAdapter(private val categoryClick: ViewHolder.CategoryClick) : RecyclerView.Adapter<CategoriesAdapter.ViewHolder>() {

	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
		val vItem = LayoutInflater
			.from(parent.context)
			.inflate(R.layout.category_item, parent, false)
		return ViewHolder(vItem, categoryClick)
	}

	override fun onBindViewHolder(holder: ViewHolder, position: Int) {
		holder.tvCategory.text = categoriesNames[position]
	}

	override fun getItemCount(): Int {
		return categoriesNames.size
	}

	class ViewHolder(vItem: View, private val categoryClick: ViewHolder.CategoryClick) : RecyclerView.ViewHolder(vItem), View.OnClickListener {
		internal val tvCategory: TextView = vItem as TextView

		init {
			tvCategory.setOnClickListener(this)
		}

		override fun onClick(tvCategory: View) {
			categoryClick.onCategoryClick(
				categoriesKeys[adapterPosition],
				categoriesNames[adapterPosition]
			)
		}

		interface CategoryClick {
			fun onCategoryClick(categoryKey: String, categoryName: String)
		}
	}

	companion object {
		private val categoriesKeys = arrayOf("all", "sightseeing", "shopping", "eating", "discovering", "playing", "traveling", "going_out", "hiking", "sports", "relaxing")

		private val categoriesNames = arrayOf("All categories", "Sightseeing", "Shopping", "Restaurants", "Museums", "Family", "Transport", "Nightlife", "Outdoors", "Sports", "Relaxation")
	}
}
